packages/reframed/src/reframed.ts

```diff
@@ -545,3 +586,21 @@ function isReframedDocument(
 ): document is ReframedDocument {
 	return "unreframedBody" in document;
 }
+
+type reframeMetadata = {
+	/**
+	 * Indicates what the value the iframe's document readyState should be.
+	 *
+	 * Note that we can't simply monkey patch this to the main document (since the iframe needs to have its own behavior regarding this state),
+	 * so we need to directly manage this state ourselves.
+	 */
+	iframeDocumentReadyState: DocumentReadyState;
+};
+
+type ReframeShadowRoot = ShadowRoot & {
+	reframeMetadata: reframeMetadata;
```

Comment:
as mentioned above, I'd rather use a private Symbol so that we don't leak the internals via the a public DOM api.

Reply:
yeah great point, done! thanks! 😄